### PR TITLE
Node Templates TF

### PR DESCRIPTION
**Pull request by zilvinasu**

### CODE REVIEW

File: castai/resource_node_template.go [hunk 1]
```diff
@@ -0,0 +1,498 @@
...
+		}
+	}
+	if err := d.Set(FieldNodeTemplateCustomLabel, flattenCustomLabel(nodeTemplate.CustomLabel)); err != nil {
+		return diag.FromErr(fmt.Errorf("setting custom label: %w", err))
+	}
+
+	return nil
+}
+
+func flattenConstraints(c *sdk.NodetemplatesV1TemplateConstraints) (map[string]interface{}, error) {
```
Comment (saumas) [inline castai/resource_node_template.go:236]: Use `any` instead of `interface{}`

Comment (zilvinasu) [reply castai/resource_node_template.go]: will do, thanks

File: examples/gke/gke_cluster_autoscaler_policies/version.tf [hunk 2]
```diff
@@ -2,6 +2,7 @@ terraform {
   required_providers {
     castai = {
       source = "castai/castai"
+      version = "0.0.0-local"
```
Comment (aldor007) [inline examples/gke/gke_cluster_autoscaler_policies/version.tf:5]: I guess this was for local testing.  Can you remove it?

**Review by saumas — APPROVED** [review]

**Review by aldor007 — APPROVED** [review]

**Comment by varnastadeus** [comment]
@zilvinasu don't forget to generate docs for the new resource

**Pull request merged**